pareto/utilities/model_modifications.py

```diff
@@ -0,0 +1,60 @@
+#####################################################################################################
+# PARETO was produced under the DOE Produced Water Application for Beneficial Reuse Environmental
+# Impact and Treatment Optimization (PARETO), and is copyright (c) 2021-2023 by the software owners:
+# The Regents of the University of California, through Lawrence Berkeley National Laboratory, et al.
+# All rights reserved.
+#
+# NOTICE. This Software was developed under funding from the U.S. Department of Energy and the U.S.
+# Government consequently retains certain rights. As such, the U.S. Government has been granted for
+# itself and others acting on its behalf a paid-up, nonexclusive, irrevocable, worldwide license in
+# the Software to reproduce, distribute copies to the public, prepare derivative works, and perform
+# publicly and display publicly, and to permit others to do so.
+#####################################################################################################
+
+###--- Imports ---###
+from importlib import resources
+from pyomo.environ import (
+    Var,
+    Binary,
+)
+
+###--- Functions ---###
+# free variables function
+def free_variables(model, exception_list=None, time_period=None):
+    for var in model.component_objects(Var):
```

Comment:
I suppose, if you use `component_data_objects` instead of `component_objects`, you need not check if the var object is indexed and then it could simplify your code. Unless you need it for another reason

Reply:
Perhaps this is so.